vernac/auto_ind_decl.ml

```diff
@@ -122,6 +122,48 @@ let check_no_indices mib =
 
 let beq_scheme_kind_aux = ref (fun _ -> failwith "Undefined")
 
+let build_beq_scheme_deps kn =
+  (* fetching global env *)
+  let env = Global.env() in
+  (* fetching the mutual inductive body *)
+  let mib = Global.lookup_mind kn in
+  (* number of inductives in the mutual *)
+  let nb_ind = Array.length mib.mind_packets in
+  (* number of params in the type *)
+  let nparrec = mib.mind_nparams_rec in
+  check_no_indices mib;
+  let make_one_eq accu i =
+    let rec aux accu c =
+      let (c,a) = Reductionops.whd_betaiota_stack Evd.empty EConstr.(of_constr c) in
+      let (c,a) = EConstr.Unsafe.(to_constr c, List.map to_constr a) in
```

Comment:
Unrelated to this PR, but what's the fix for this `Unsafe` use in the file?

Reply:
I think that we could simply use full kernel βιδ from Reductionops.

Reply:
Yeah so I'd be great if we add a comment for the day a c̶r̶a̶z̶y̶ bold person wants to refactor this module.

Reply:
Done.